server/helpers/adapters_tracker.go

```diff
@@ -179,3 +213,91 @@ func (a *AdaptersTracker) UndeployAdapter(ctx context.Context, adapter models.Ad
 	a.RemoveAdapter(ctx, adapter)
 	return nil
 }
+
+func int32Ptr(n int32) *int32 {
+	return &n
+}
+
+// SetOverrideValues returns the value overrides to install/upgrade helm chart
+func (a *AdaptersTracker) setOverrideValues(adapterName string, exposedPort int32, enable bool) map[string]interface{} {
+	// Initialize the valueOverrides with default values and set config for selected adapter
+	valueOverrides := map[string]interface{}{}
+	for _, availableAdapter := range models.ListAvailableAdapters {
+		valueOverrides[availableAdapter.Name] = map[string]interface{}{
+			"enabled": false,
+		}
+		if enable && availableAdapter.Name == adapterName {
+			valueOverrides[availableAdapter.Name] = map[string]interface{}{
+				"enabled": true,
+				"service": map[string]interface{}{
+					"name":      adapterName,
+					"namespace": core.MesheryNamespace,
+					"spec": map[string]interface{}{
+						"selector": map[string]string{
+							"app": adapterName,
+						},
+						"ports": []map[string]interface{}{
+							{
+								"name":     "gRPC",
+								"protocol": "TCP",
+								"port":     exposedPort,
+							},
+						},
+						"type": "ClusterIP",
```

Comment:
Instead of building this out by hand, so to speak, is there no existing struct / helm function to reference?